aten/src/ATen/native/sparse/SparseCsrTensorMath.cpp

```diff
@@ -60,6 +70,29 @@ void convert_indices_from_coo_to_csr_cpu(const Tensor& result, const Tensor& inp
     data_out[i] = static_cast<output_t>(numel);
 }
 
+template <typename input_t, typename output_t>
+void convert_indices_from_csr_to_coo_cpu(const Tensor& result, const Tensor& crow_indices, const Tensor& col_indices) {
+  int64_t nrows = crow_indices.numel() - 1;
+  if (nrows == 0) {
+    result.zero_();
+    return;
+  }
+  auto crow_indices_ = crow_indices.expect_contiguous();
+  auto col_indices_ = col_indices.expect_contiguous();
+  const input_t* crow_indices_data_in = crow_indices_->data_ptr<input_t>();
+  const input_t* col_indices_data_in = col_indices_->data_ptr<input_t>();
+  TORCH_INTERNAL_ASSERT(result.is_contiguous());
```

Comment:
@pearu - I'm not convinced this *actually* makes sure the indices are contiguous, but it's not possible to construct a csr Tensor with non-contiguous indices. The user would have to really want it. Maybe we can dig deeper into this as future work?

Reply:
`result` here is a regular dense strided tensor.
When constructing sparse csr matrix the input is validated using `_validate_sparse_csr_tensor_args` function and contiguity is the first thing being checked https://github.com/pytorch/pytorch/blob/907e587370a72979eb97c4723c57a7559b772511/aten/src/ATen/native/sparse/SparseCsrTensor.cpp#L23-L35

Reply:
Got it, result here are the indices to then be used in construction of the COO Tensor. Maybe we should change the name, but it's not pressing.

For the csr tensor args, while the constructor validates the indices are both contiguous, a user could modify them in-place to be non-contiguous (even though we could say that it's their own fault then). I do still think it raises an interesting question whether "is_contiguous" on CSR Tensors should take the contiguity of the indices into account.   This doesn't seem important though until we actually write kernels that could potentially benefit from that.